dali/operators/sequence/element_extract.h

```diff
@@ -129,16 +127,13 @@ class ElementExtract : public Operator<Backend> {
  private:
   std::vector<int> element_map_;
 
-  struct NoOpPlaceholder {
-    explicit NoOpPlaceholder(size_t) {}
-  };
-
-  // Do not use ScatterGather as a member for CPU implementation
   std::conditional_t<
-      std::is_same<Backend, GPUBackend>::value,
-      kernels::ScatterGatherGPU,
-      NoOpPlaceholder> scatter_gather_;
-  static constexpr size_t kMaxSizePerBlock = 1 << 18;  // 256 kB per block
+      std::is_same<Backend, CPUBackend>::value,
+      kernels::ScatterGatherCPU,
+      kernels::ScatterGatherGPU> scatter_gather_;
+  // 1 MB per block for CPU, 256 kB per block for GPU
+  static constexpr size_t kMaxSizePerBlock =
+      std::is_same<Backend, CPUBackend>::value ? 1 << 20 : 1 << 18;
```

Comment:
Why would we like to have _larger_ blocks on CPU?

Reply:
Just guessing. I think I will go with trying at least `number of threads * 3` tasks, and for the small buffer without the thread pool.